iaas-common/src/main/java/io/github/cloudiator/persistance/PricingModelRepositoryJpa.java

```diff
@@ -0,0 +1,81 @@
+package io.github.cloudiator.persistance;
+
+import com.google.inject.Provider;
+import com.google.inject.TypeLiteral;
+import de.uniulm.omi.cloudiator.domain.OperatingSystem;
+import de.uniulm.omi.cloudiator.sword.domain.Api;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.Query;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class PricingModelRepositoryJpa extends BaseModelRepositoryJpa<PricingModel> implements PricingModelRepository {
+    @Inject
+    public PricingModelRepositoryJpa(Provider<EntityManager> entityManager, TypeLiteral<PricingModel> type) {
+        super(entityManager, type);
+    }
+
+    @Nullable
+    @Override
+    public PricingModel findByCloudUniqueId(String cloudUniqueId) {
+        checkNotNull(cloudUniqueId, "cloudUniqueId is null");
+        String queryString = String
```

Comment:
Could be moved to constructor to avoid creating multiple instances of this query.

Reply:
done